sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/generators.scala

```diff
@@ -85,8 +85,10 @@ case class UserDefinedGenerator(
   override protected def makeOutput(): Seq[Attribute] = schema
 
   override def eval(input: Row): TraversableOnce[Row] = {
+    // TODO(davies): improve this
+    val input_schema = StructType(children.map(e => StructField(e.simpleString, e.dataType, true)))
```

Comment:
Can you explain what is the problem at here? Why we need to do the get the input schema in the `eval`? Also, `inputSchema` seems better than `input_schema`.